foundationdb-gen/src/bin/foundationdb-options-gen.rs

```diff
@@ -0,0 +1,332 @@
+extern crate inflector;
+extern crate xml;
+#[macro_use]
+extern crate error_chain;
```

Comment:
Failure is quite nice and simple. For the generator, you can get away with just using `format_err!` and returning the `failure::Error` type. in the result type, example:

```rust
fn example() -> Result<Type, failure::Error> {
   Err(format_err!("my failure message: {}", extra_info)
}
```

Can we switch to that instead? I believe most of Rust is going to be going in this direction.

Reply:
I changed it to `failure`.